### ISSUE TITLE
About diffu-grpo code

### ISSUE THREAD

**Issue by dllmcool** [dllm-reasoning/d1#10]
Hi,
Thanks for the great work! I noticed that #training steps is specified in the appendix. However, it seems that there is no stopping mechanism in the diffu-grpo code (i.e., stop after certain steps). 
- Do you mannually stop the code?
- What is the exact steps of the best performing checkpoint for each task (Since in the paper it seems that you evaluated differnt checkpoints at different steps and report the best one)?
Thanks

**Comment by devaansh100** [comment]
Hi, thanks for your interest! 
We do manually stop the code when the reward seems to have converged.

We will get back with the exact step counts after collating that data.

**Comment by dllmcool** [comment]
Thanks for clarifying the manual stopping.
Regarding exact step count, I was wondering if this will take much effort to report. Based on my understanding, you should have recorded which checkpoint you are using to get each results, and reporting this might not require much time. I'm reaching out for step count as I'm not able to reproduce the results simply with the checkpoint at last step.
Thanks!

**Comment by devaansh100** [comment]
These numbers are stored on meta's clusters due to which is why it might take some time to get and report them. What you can do in the meantime is get the generations from various checkpoints under the three generation length settings (using `run_eval.sh`), followed by aggregation (using `parser_json.py`). From these, pick the best one.

As already mentioned in the paper, evaluating at the last step would not reproduce the results in Table 1 (check image). Also note that evaluation is not always completely deterministic (due to various factors out of our control, but you should be able to reproduce them within reasonable error bars!)

<img width="968" alt="Image" src="https://github.com/user-attachments/assets/33c4bd8e-bd59-4af5-8b97-f7263b35e189" />

This evaluation is due to the unstableness of RL training. If you also check our latest arxiv version, we have added experiments on evaluating just a single checkpoint on unified training (training all tasks together) and coding tasks. Code for this is coming soon too!

**Comment by STAR-CV** [comment]
hi, great work! I'd like to ask if there are any approximate results for the optimal model's step count range at this time? @devaansh100
